shapely/geometry/multilinestring.py

```diff
@@ -132,6 +132,10 @@ def geos_multilinestring_from_py(ob):
     # add to coordinate sequence
     for l in range(L):
         geom, ndims = linestring.geos_linestring_from_py(obs[l])
+
+        if lgeos.GEOSisEmpty(geom):
+            raise ValueError("Can't create MultiLineString with empty component")
```

Comment:
@tomplex what would you think about a more shapely-specific exception here? Like a new shapely.errors.EmptyPartError?

I got this idea from Brett Slatkin: https://books.google.com/books?id=bTUFCAAAQBAJ&pg=PA185&lpg=PA185&dq=slatkin+python+base+exception&source=bl&ots=GahcQjraMD&sig=ACfU3U15-chIf93EZAganyYHVhWGQpI_EA&hl=en&sa=X&ved=2ahUKEwiG_tj4korpAhUOAp0JHUleDVEQ6AEwBXoECAsQAQ#v=onepage&q=slatkin%20python%20base%20exception&f=false

Reply:
That would be fine with me, but you mentioned in #836 you'd rather this be a warning and skip the empty geometry instead of `raise`-ing. Would you prefer if I re-worked this to do that instead?

Reply:
After more thought, I think an exception is the right approach. GEOS is going to crash otherwise, so an exception only helps.